Add some PostgreSQL compatibility features mentioned in issue #2450

auntyellow opened the pull request:
- SET client_min_messages
- SET client_encoding
- SHOW ssl
- set_config()
- pg_relation_size()
- obj_description()
- column "pg_catalog"."proargtypes"
- ::regclass (use function alias "pg_get_oid" when DATABASE_TO_LOWER=TRUE)

katzyn commented on h2/src/main/org/h2/command/Parser.java:
```diff
@@ -1568,6 +1568,9 @@ private Prepared parseShow() {
             buff.append("'UTF8' AS SERVER_ENCODING FROM DUAL");
+        } else if (readIf("SSL")) {
+            // for PostgreSQL compatibility
+            buff.append("'off' AS SSL FROM DUAL");
```
Please, merge changes from the `master` branch and use `"'off' SSL"` here.

auntyellow replied:
merged

katzyn commented on h2/src/main/org/h2/command/Parser.java:
```diff
@@ -4657,7 +4660,8 @@ private Expression readTerm() {
             if (readIf("REGCLASS")) {
-                FunctionAlias f = findFunctionAlias(database.getMainSchema().getName(), "PG_GET_OID");
+                FunctionAlias f = findFunctionAlias(database.getMainSchema().getName(),
+                        identifiersToUpper ? "PG_GET_OID" : "pg_get_oid");
```
`database.sysIdentifier()` should be used here, but I'm not sure about the whole such replacement with this function. It doesn't have any real implementation and it isn't defined when PG server is not in use.

auntyellow replied:
fixed

katzyn commented on h2/src/main/org/h2/server/pg/pg_catalog.sql:
```diff
@@ -149,6 +149,7 @@ create table pg_catalog.pg_proc(
     oid int,
     proname varchar_ignorecase,
     prorettype int,
+    proargtypes array,
```
`int array` or some other normal data type should be used here.

auntyellow replied:
fixed

katzyn commented on h2/service/wrapper.conf:
```diff
@@ -1,101 +1,101 @@
+#********************************************************************
+# Wrapper Properties
+#********************************************************************
+# Java Application
```
You can't add this configuration file that easy. You need to adjust generation of Windows and ZIP distributions too, because they are currently packaged with another wrapper. It's better to do it in a separate pull request.

See also #2410.

auntyellow replied:
Sorry, I make some mistakes duriing merge. This file should not be commited ;-(

I will try rebase and double check before push.

auntyellow committed:
Add some PostgreSQL compatibility features mentioned in issue #2450

SET client_min_messages
SET client_encoding
SHOW ssl
set_config()
pg_relation_size()
obj_description()
column "pg_catalog"."proargtypes"
::regclass (use function alias "pg_get_oid" when DATABASE_TO_LOWER=TRUE)

katzyn commented:
Thanks!
